src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1058,31 +1322,126 @@ def _set_up_defaults(
+        connected_env_list = list_connected_environments(cmd=cmd, resource_group_name=resource_group.name)
+        env_list = []
+        for e in connected_env_list:
+            if env.name and env.name != e["name"]:
+                continue
+            if location and format_location(e["location"]) != location:
+                continue
+            if custom_location.name and e["extendedLocation"]["name"].lower() != custom_location.get_rid().lower():
+                continue
+            if custom_location.connected_cluster_id:
+                custom_location_from_env = get_custom_location(cmd=cmd, custom_location_id=e["extendedLocation"]["name"])
+                if custom_location_from_env is None or custom_location.connected_cluster_id.lower() != custom_location_from_env.host_resource_id.lower():
+                    continue
+            env_list.append(e)
+
+        if len(env_list) == 1:
+            if not resource_group.name:
+                resource_group.name = parse_resource_id(env_list[0]["id"])["resource_group"]
+            env.set_name(env_list[0]["id"])
+            env.custom_location_id = env_list[0]["extendedLocation"]["name"]
+        if len(env_list) > 1:
+            if env.name:
+                raise ValidationError(
+                    f"There are multiple Connected Environments with name {env.name} on the subscription. "
+                    "Please specify which resource group your Connected environment is in."
+                )  # get ACR details from --image, if possible
+            else:
+                if not resource_group.name:
+                    resource_group.name = parse_resource_id(env_list[0]["id"])["resource_group"]
+                env.set_name(env_list[0]["id"])
+                env.custom_location_id = env_list[0]["extendedLocation"]["name"]
+
+
+def _infer_existing_custom_location_or_extension(
+        cmd,
+        name,
+        location,
+        resource_group: "ResourceGroup",
+        env: "ContainerAppEnvironment",
+        custom_location: "CustomLocation",
+        extension: "Extension"
+):
+    # Try to get existed custom location and extension
+    # Set up default values for not existed resources(env, custom location, extension)
+    if env.is_connected_environment():
+        if not env.check_exists():
+            _prepare_custom_location_args(cmd, custom_location=custom_location, extension=extension)
+            if location is None:
+                env.location = custom_location.location
+                resource_group.location = custom_location.location
+            if custom_location.exists:
+                env.custom_location_id = custom_location.get_rid()
+            else:
+                random_int = randint(0, 9999)
+                resource_group.name = resource_group.name if resource_group.name else get_randomized_name(get_profile_username(), random_int=random_int)
+                env.name = env.name if env.name else "{}-env".format(name).replace("_", "-")
```

Comment:
could random int code put in get_randomied_name function?

Reply:
We hope to use same int code in resource_group_name, env_name, custom_location_name. So cannot put in get_randomied_name function, it will generate random name and cannot be consistent.